src/libmongoc/tests/test-mongoc-background-monitoring.c

```diff
@@ -41,7 +41,8 @@ typedef struct {
 typedef enum {
    TF_FAST_HEARTBEAT = 1 << 0,
    TF_FAST_MIN_HEARTBEAT = 1 << 1,
-   TF_AUTO_RESPOND_POLLING_HELLO = 1 << 2
+   TF_AUTO_RESPOND_POLLING_HELLO = 1 << 2,
+   TF_NO_MONGODB_API_VERSION = 1 << 3, /* if set, do not pick up the MONGODB_API_VERSION environment variable */
```

Comment:
I suggest removing `TF_NO_MONGODB_API_VERSION` and replacing the calls to `mock_server_receives_legacy_hello` with `mock_server_receives_any_hello`.

That will enable `tf_new` to use `test_framework_client_pool_new_from_uri`.